maven-surefire-common/src/main/java/org/apache/maven/plugin/surefire/extensions/SurefireForkChannel.java

```diff
@@ -129,7 +159,7 @@ private final void setTrueOptions( SocketOption<Boolean>... options )
     @Override
     public String getForkNodeConnectionString()
     {
-        return "tcp://" + localHost + ":" + localPort;
+        return "tcp://" + localHost + ":" + localPort + ( sessionId == null ? "" : "?sessionId=" + sessionId );
```

Comment:
If this is local host, why do we need it as a vaialble?

Reply:
We do not know if it is localhost. Nothing must be hard coded in this place. Only one point in whole `SurefireForkChannel` can do it and it is the constructor at the [Line 93](https://github.com/apache/maven-surefire/blob/ea4020701cb2c53545a0ae3a2820938195f4077f/maven-surefire-common/src/main/java/org/apache/maven/plugin/surefire/extensions/SurefireForkChannel.java#L93). We provide abstraction and we provide two default implementations but it does not mean that later it would not be foreign IP address since some user inherited this class overriding the constructor. This reimplementation can be done in the POM `<forkNode implementation=""/>`. This connection string goes to the properties file and to the fork JVM.

Reply:
It's the best programming approach to set the value in one place of the class and reference the value even if it would be the same with hard coded value. We do it with our experiences that the analysis is simple and easy to change. Therefore our suumptions of a constant string `locahost` does not end up in 3 places, nothing but only one. It's later easy to manipulate with such a class and extend it with alternative construcotr, etc. These are the practices which drive us to this design.

Reply:
I see, but there a few problems here if you want to have maximum flexibility:
* Just use variables `host` + `port`. Don't imply that you are limited to `localhost` even if you use it only
* Don't use `Inet4Address`. If the host is running on IPv6 only, the code will fail. Use `InetAddress` and it will work for both.

Reply:
@michael-o used InetAddress.